src/Components/PhoneNumberInput/validatePhoneNumber.ts

```diff
@@ -0,0 +1,61 @@
+import { debounce } from "lodash"
+import { fetchQuery, graphql } from "react-relay"
+import { createRelaySSREnvironment } from "System/Relay/createRelaySSREnvironment"
+import { validatePhoneNumberQuery } from "__generated__/validatePhoneNumberQuery.graphql"
+
+const relayEnvironment = createRelaySSREnvironment()
+
+export type PhoneNumber = {
+  national: string
+  regionCode: string
+}
+
+const validator = debounce(
+  async (
+    { national, regionCode }: PhoneNumber,
+    resolve: (value: boolean) => void
+  ) => {
+    if (!national || national.length < 5 || !regionCode) {
+      return resolve(false)
+    }
+
+    try {
+      const response = await fetchQuery<validatePhoneNumberQuery>(
+        relayEnvironment,
+        graphql`
+          query validatePhoneNumberQuery(
+            $phoneNumber: String!
+            $regionCode: String
+          ) {
+            phoneNumber(phoneNumber: $phoneNumber, regionCode: $regionCode) {
+              isValid
+            }
+          }
+        `,
+        { phoneNumber: national, regionCode }
+      ).toPromise()
+
+      if (!response?.phoneNumber) {
+        // Assume the phone number is valid if we can't validate it
+        return resolve(true)
+      }
+
+      return resolve(!!response.phoneNumber.isValid)
+    } catch (err) {
+      console.error(err)
+
+      // Assume the phone number is valid if we can't validate it
+      return resolve(true)
```

Comment:
Just note that on errors I'm simply assuming it's valid. IMO we don't want to prevent form submission in the case of error and can probably just safely assume it's entered correctly.